test/jdk/java/awt/List/MouseDraggedOriginatedByScrollBarTest.java

```diff
@@ -77,27 +82,62 @@ private static Frame createTestUI() {
             new MouseMotionAdapter(){
                 @Override
                 public void mouseDragged(MouseEvent me){
-                    PassFailJFrame.log(me.toString());
+                    System.out.println(me.toString());
+                    throw new RuntimeException("Test failed. Mouse dragged " +
+                            "event detected.");
                 }
             });
 
         list.addMouseListener(
             new MouseAdapter() {
                 public void mousePressed(MouseEvent me) {
-                    PassFailJFrame.log(me.toString());
+                    System.out.println(me.toString());
+                    throw new RuntimeException("Test failed. Mouse pressed " +
+                            "event detected.");
                 }
 
                 public void mouseReleased(MouseEvent me) {
-                    PassFailJFrame.log(me.toString());
+                    System.out.println(me.toString());
+                    throw new RuntimeException("Test failed. Mouse released " +
+                            "event detected.");
                 }
 
                 public void mouseClicked(MouseEvent me){
-                    PassFailJFrame.log(me.toString());
+                    System.out.println(me.toString());
+                    throw new RuntimeException("Test failed. Mouse clicked " +
+                            "event detected.");
                 }
             });
 
         frame.setLayout(new FlowLayout());
         frame.pack();
-        return frame;
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    private static void test() throws Exception {
+        Robot robot;
+        try {
+            robot = new Robot();
+        } catch (Exception ex) {
+            throw new RuntimeException("Can't create robot");
+        }
+        robot.waitForIdle();
+        robot.delay(1000);
+        robot.setAutoWaitForIdle(true);
+
+        // Focus default button and wait till it gets focus
+        EventQueue.invokeAndWait(() -> {
+            loc = list.getLocationOnScreen();
+            width = list.getWidth();
+        });
+        robot.mouseMove(loc.x + width - 10, loc.y + 20);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        for (int i = 0; i < 30; i++) {
+            Point p = MouseInfo.getPointerInfo().getLocation();
+            robot.mouseMove(p.x, p.y + 1);
+        }
```

Comment:
```suggestion
        for (int i = 0; i < 30; i++) {
            robot.mouseMove(loc.x, loc.y + 1);
        }
```
You may copy `loc` into a non-volatile local variable (micro-optimisation).

Reply:
Sure, thanks for the suggestion!